blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md

```diff
@@ -0,0 +1,905 @@
+        delete glasses[glassLength - 1];
+        glassLength--;
+
+        emit deleteGadgetEvent(_index);
+    }
+```
+
+In this session, we have a function called `removeGadget` which allows the owner of a gadget to delete it from the system by passing in an index number.
+
+When someone calls the removeGlass function, we first check to make sure that the person calling the function is the owner of the gadget they want to delete. If the condition is not met, the function stops executing and returns an error message saying `Only owner can delete`.
+
+Assuming the person calling the function is the owner of the gadget, we proceed to remove the gadget from the system. We do this by copying the last gadget in the system to the location of the gadget being deleted, and then deleting the last gadget. After that, we decrement the length of the gadget array by 1 to reflect the removal of the gadget.
+
+Once we have successfully removed the gadget from the system, we emit a `deleteGadgetEvent` event to let other parts of the system know that a gadget has been deleted.
+
+Next we add the `buyGadget` function
+
+```solidity
+ function buyGlass(uint _index) public payable {
+        Glass storage currentGlass = glasses[_index];
+        require(
+            currentGlass.owner != msg.sender,
+            "You can't buy your own glass"
+        );
+        require(
+            IERC20Token(cUsdTokenAddress).transferFrom(
+                msg.sender,
+                glasses[_index].owner,
+                glasses[_index].price
+            ),
+            "Transfer failed."
+        );
+
+        address seller = glasses[_index].owner;
+
+        glasses[_index].owner = payable(msg.sender);
+
+        emit buyGadgetEvent(seller, msg.sender, _index);
+    }
+```
+
+This code snippets defines a function called buyGadget, which allows a user to purchase a gadget from the system by passing in an index number and paying the price of the gadget in cUSD (a type of cryptocurrency).
+
+When someone calls the `buyGadget` function, it first creates a reference to the gadget being purchased using the Glass storage currentGlass = glasses[_index] statement.
+
+Next, the function checks to make sure that the person calling the function is not the owner of the gadget. If the condition is not met, the function stops executing and returns an error message saying "You can't buy your own gadget".
+
+Assuming the person calling the function is not the owner of the gadget, the function then attempts to transfer the price of the gadget in cUSD from the buyer to the owner of the gadget using the `IERC20Token(cUsdTokenAddress)`.transferFrom statement. If the transfer is successful, the function updates the owner of the gadget to be the buyer, and emits a buyGadgetEvent event to let other parts of the system know that a gadget has been purchased.
+
+Finally, for the smart contract we add the `Like` and the getGadgetLength` function.
```

Comment:
fix the formating for this line, there's no openning tag for getGadgetLength function